src/applications/common/schemaform/fields/ArrayField.jsx

```diff
@@ -44,6 +45,19 @@ export default class ArrayField extends React.Component {
     this.scrollToRow = this.scrollToRow.bind(this);
   }
 
+  componentWillMount() {
+    const { schema, uiSchema, formData = [], registry } = this.props;
+    const items = (formData && formData.length)
+      ? formData
+      : [getDefaultFormState(schema, undefined, registry.definitions)];
+    const lastItem = items[items.length - 1];
+    this.showInViewMode = !!(uiSchema['ui:options'].showLastItemInViewMode && lastItem);
+    const lastIndex = this.props.formData && this.props.formData.length - 1;
+    if (this.props.errorSchema[lastIndex] && !errorSchemaIsValid(this.props.errorSchema[lastIndex])) {
+      this.showInViewMode = false;
+    }
+  }
```

Comment:
@cvalarida this change fixes the bug (view changing once user input is valid) you found.

Reply:
Does this cover the case where hitting 'edit' under any circumstance advances to the next page? Cause I ran into that. Basically -> use existing, valid initialData -> hit 'Edit' next to the first (and only) item -> advances to payments page.